.gitignore

```diff
@@ -1,4 +1,5 @@
 target/
+Cargo.lock
```

Comment:
I haven't seen this before, but just found the following resource on removing `Cargo.lock`

Is this change for the same rationale?
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
I didn't know it is recommended to check in the .lock file for binaries, I never checked it in. But it works either way and in general, the Cargo.lock is not checked in. But apparently it can be used to determine the exact dependencies a build was compiled with.

Reply:
Actually if you don't check in the `Cargo.lock` for a binary, the build will in general not be repeatable, because the versions of dependencies could have been updated. If you have a dependency with `dependency = "x.y.z"` in your `Cargo.toml`, cargo will take any version that has the same major number `x.y` and at least `z` minor version. You will need to specify `"=x.y.z` to get the exact version, however this is rarely done.